### DISCUSSION TITLE
Patched for GFX7xx?

### DISCUSSION THREAD

**Discussion by iamhumanipromise** [rocm-arch/rocm-arch#838] [Support] [answered]
Hello! rocminfo displays my gfx701 device with the full amdgpu-amd-amdhsa--gfx701 identifier. opencl however begins to show the only the first part of clinfo before freezing. 

Is there a way to patch this to force gfx701 support? (In a similar fashion to the arch4edu. llvm-amdgpu package, I suppsose)

It looks like the device libs have bitcode for gfx701. 

(Use case: using multiple gfx701 devices in a distcc system to speed up compile times while using new GPUs for other projects with -fopemp offload --- also to play with opencl performance tests)

**Comment by petronny** [comment]
`llvm-amdgpu` has been dropped from the arch4edu repository on github.
But I haven't got time to remove it (and other dropped and old packages) from the pacman repository yet.

**Reply by iamhumanipromise** [reply]
Please keep it, if possible, for older cards, as part of a "legacy bundle"?

**Reply by tpkessler** [reply]
With our packages we try to follow upstream as close as possible. A legacy version of ROCm (that is pre 2.0) would require a separate repo.

**Reply by iamhumanipromise** [reply]
Ah that is unfortunate. What is needed for a legacy version to exist? Would like to provide compute resources or testing, maybe even host (Ubiquiti UDM + 400/40mb connection, would cap upload speeds however)

**Reply by acxz** [reply]
The difficulty in trying to get ROCm pre 2.0 to build is so large than any amount of resources you provided, short of hiring a full time compile/integration engineer, would be insufficient. If you really need this and have the money/resources to spare it is best to just upgrade your hardware.

**Comment by acxz** [comment]
What was the patch for the llvm-amdgpu package? If it doesn't interfere with support for other gfx's then feel free to submit a PR and we can take a look at it.

**Reply by tpkessler** [reply]
I cannot recall that there was a patch for llvm-amdgpu for Hawaii.

**Reply by iamhumanipromise** [reply]
Apologies, it appears it was an unofficial script "patch" for Fiji/Tonga cards which are GFX8xx cards. Due to the naming nomenclature (R9 285) and co-existance on the marketplace, I mistakenly believed Hawaii to be GCN 3.0 and have been corrected.

Thank you for your continued efforts.

**Comment by tpkessler** [comment, marked as answer]
I doubt that it is possible to support (pre-) Hawaii cards with current ROCm. Support was dropped before ROCm 2.0, see [this comment](https://github.com/RadeonOpenCompute/ROCm/issues/871#issuecomment-976169575) and the discussion in the linked issue.
